builds/jenkins-build-sls-app/Jenkinsfile

```diff
@@ -0,0 +1,1062 @@
+
+      withCredentials([
+        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: configLoader.AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']
+      ]) {
+        try {
+          // initialize aws credentials
+          def randomString = utilModule.generateRequestId();
+          def credsId = "jazz-${randomString}";
+          sh "aws configure set profile.${credsId}.region ${configLoader.AWS.REGION}"
+          sh "aws configure set profile.${credsId}.aws_access_key_id $AWS_ACCESS_KEY_ID"
+          sh "aws configure set profile.${credsId}.aws_secret_access_key $AWS_SECRET_ACCESS_KEY"
+
+          // Generate serverless yml file with domain added in function name
+          echo "Generate deployment env with domain"
+
+          if (isScheduleEnabled) {
+            def eventsArns = getEventsArn(config, environment_logical_id)
+            for (def i = 0; i < eventsArns.size(); i++) {
+              def arn = eventsArns[i]
+              events.sendCompletedEvent('CREATE_ASSET', null, utilModule.generateAssetMap("aws", arn, "cloudwatch_event", config), environment_logical_id);
+            }
+          }
+
+          def envBucketKey = "${env_key}${configLoader.JAZZ.S3_BUCKET_NAME_SUFFIX}"
+          echoServerlessFile()
+          def deployOutput = servelessDeploy(environment_logical_id, envBucketKey)
+
+// Shoving deploy output lines into array
+          def deployLines = deployOutput.split('\n')
+          def resourceArns = []
+          def afterStackOutputs = false
+// Extracting all the lines that follow 'Stack Outputs'. These lines are supposed to contain arns of resources created
+          for(def line: deployLines) {
+            if(afterStackOutputs) resourceArns.add(line)
+            if(line.contains('Stack Outputs')) afterStackOutputs = true // Here we are only interested in lines that follow 'Stack Outputs' marker
+          }
+
+          for(def name2ArnString: resourceArns) {
+            def name2ArnPair = name2ArnString.split(': ') // Here we rely on the space after the ': ' so it is split properly without affecting the arn itself that is full of colons but never has a space after
+            if(name2ArnPair.size() > 1) {
+              def arn = name2ArnPair[1].trim()
+              if(arn.startsWith('arn')) {
+                 def arnAsArray = arn.split(':') // Here we splitting the arn itself in hope to obtain type of the resource. Should be the third element: arn:aws:dynamodb:us-east-1:102707241671:table/jazzUsersTable53
+                 if(arnAsArray.size() > 2) { // Making sure that the third element exists
+                   def artifactType = arnAsArray[2]
+                   if(artifactType == 'kinesis') artifactType = 'kinesis_stream' // 'kinesis_stream' needs to be instaead of simple 'kinesis' that did not get persisted :-( by itself
```

Comment:
Why is this required?

Reply:
To create the assets I need to call events.sendCompletedEvent(...) method that requires 'arn' and 'resource type'. So I spike the serveless.yml file with 'Output' section here:
```groovy
      echo "resourceKeys => $resourceKeys"
      /* Forming a record that extracts the arn from resulting item for each of resource key extracted from resources:
      UsersTableArn:\n
        Value:\n
          \"Fn::GetAtt\": [ usersTable, Arn ]\n
      */
      resourceKeys.collect{name ->
          bigOutputsElem[name+'Arn']=["Value":["Fn::GetAtt":[name, "Arn"]]]
      }
```
and then I expect that the desired information about resources created will appear inside the 'serverless deploy' command output. The interesting fact that I also noticed is that the 'resource type' I can extract from 'arn' itself. So what I have tried to achieve here is to traverse through the output returned by the 'serverless deploy' command. To skip the part of output with I deem to be irrelevant I use the following piece of code that is patiently waiting for that magic 'Stack Outputs' line. 
```groovy
          for(def line: deployLines) {
            if(afterStackOutputs) resourceArns.add(line)
            if(line.contains('Stack Outputs')) afterStackOutputs = true // Here we are only interested in lines that follow 'Stack Outputs' marker
          }
```
Then I expect the records in the format as follows "<resource_name>: <arn>" like that:
myDynamoTable: arn:aws:dynamodb:us-east-1:102707241671:table/jazzUsersTable53 

The easiest way to extract the type is to parse the arn and extract the third element from it and that exactly the reason behind 
```groovy
def artifactType = arnAsArray[2]
```
That works well except the kinesis stream where I had to resort to the if-statement
```groovy
  if(artifactType == 'kinesis') artifactType = 'kinesis_stream'
```
I did not see any simpler method to extract the resource type